### PR TITLE
Removed dependency on cloud-diagnostics-common

### PR DESCRIPTION
**Pull request by kjin**
Removed all dependencies on cloud-diagnostics-common, replacing with equivalent functionality in cloud-common.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,7 +126,18 @@ var publicAgent = {
     if (!config.enabled) {
       return this;
     }
-    var logger = common.logger.create(config.logLevel, '@google/cloud-trace');
+
+    var logLevel = config.logLevel || 2;
```
Comment (ofrobots) [inline index.js]: I think `initConfig` and config.js guarantee that config.logLevel will be defined by this point. At any rate, we shouldn't duplicate the default logLevel that is also in the config.

Aside: should the default logLevel in config.js continue to be an integer, or should it be `'error'`.

File: index.js [hunk 2]
```diff
@@ -125,7 +126,18 @@ var publicAgent = {
...
+    if (logLevel < 0) {
```
Comment (ofrobots) [inline index.js]: We have a hazard here if `config.logLevel` came from the environment. Numbers would show up as strings. That's not fatal however since the expression `'4' < 0` does evaluates correctly due to type coersion `false`.

However `'monkey' < 0` is a valid comparison too that evaluates to `false`. In other words, if the user had an invalid log level, we disable logging for them.

File: index.js [hunk 3]
```diff
@@ -125,7 +126,18 @@ var publicAgent = {
...
+      logLevel = 0;
+    } else if (logLevel > 4) {
```
Comment (ofrobots) [inline index.js]: hard-coded constant

File: index.js (hunk 1 above)
Comment (kjin) [inline index.js]: Is the specification of `logLevel` is not baked in? If not, I think having `logLevel` accept the name of the level makes more sense and I'd be all for it - though it would involve changing across the other agents as well to preserve uniformity.

Comment (ofrobots) [inline index.js]: baked-in where?

Comment (kjin) [inline index.js]: As in, whether it's set in stone that it should be a number rather than the log level itself.

**Comment by kjin** [comment]
Addressed inconsistencies in log levels here: https://github.com/GoogleCloudPlatform/google-cloud-node/pull/1900

File: index.js [hunk 4]
```diff
@@ -74,7 +75,11 @@ var initConfig = function(projectConfig) {
   util._extend(config, require('./config.js').trace);
   util._extend(config, projectConfig);
   if (process.env.hasOwnProperty('GCLOUD_TRACE_LOGLEVEL')) {
-    config.logLevel = process.env.GCLOUD_TRACE_LOGLEVEL;
+    try {
+      config.logLevel = parseInt(process.env.GCLOUD_TRACE_LOGLEVEL);
+    } catch (e) {
+      // Just use the original log level if GCLOUD_TRACE_LOGLEVEL is malformed.
```
Comment (AdriVanHoudt) [inline index.js]: why not print out some sort of warning because this is probably not what the user wants right?

Comment (kjin) [inline index.js]: Makes sense - though the logger isn't initialized at this point (since it depends on the value of `config.logLevel`).

Do you think it'd make sense to defer printing of the warning until after the logger is initialized?

Comment (AdriVanHoudt) [inline index.js]: Not familiar with internals but makes sense to defer when the logger isn't there.
I assume this is not breaking anything just failsafe. So just letting the user know (even a bit later) seems fine

File: index.js [hunk 5]
```diff
@@ -121,11 +108,47 @@ var publicAgent = {
...
+    var logLevelValid = true;
+
+    // Initialize config object
+    var util = require('util');
+    var config = {};
+    util._extend(config, require('./config.js').trace);
+    util._extend(config, projectConfig);
+    if (process.env.hasOwnProperty('GCLOUD_TRACE_LOGLEVEL')) {
+      if (process.env.GCLOUD_TRACE_LOGLEVEL.match(/^\d+$/)) {
+        config.logLevel = parseInt(process.env.GCLOUD_TRACE_LOGLEVEL);
```
Comment (ofrobots) [inline index.js]: nit: always specify a radix for parseInt.

File: index.js [hunk 6]
```diff
@@ -121,11 +108,47 @@ var publicAgent = {
...
-    var config = initConfig(projectConfig);
+    var logLevelValid = true;
+
+    // Initialize config object
+    var util = require('util');
+    var config = {};
+    util._extend(config, require('./config.js').trace);
+    util._extend(config, projectConfig);
+    if (process.env.hasOwnProperty('GCLOUD_TRACE_LOGLEVEL')) {
+      if (process.env.GCLOUD_TRACE_LOGLEVEL.match(/^\d+$/)) {
```
Comment (ofrobots) [inline index.js]: Could you avoid this by doing `parseInt` and checking if the result is `NaN`?

File: index.js [hunk 7]
```diff
@@ -121,11 +109,46 @@ var publicAgent = {
...
+      logLevelValid = false;
+    }
+    var logger = common.logger({
+      level: common.logger.LEVELS[logLevel],
+      tag: '@google/cloud-trace'
+    });
+    if (!logLevelValid) {
+      var desiredLogLevel = process.env.GCLOUD_TRACE_LOGLEVEL ||
+        config.logLevel;
+      logger.warn('Provided log level ' + desiredLogLevel + ' is invalid, ' +
```
Comment (ofrobots) [inline index.js]: I think you can use `console.error` here to avoid issues with the default logLevel not allowing `warn` to actually be printed. This would be a valid use of `console.error`.

This should also simplify the logic here as you wouldn't need to carry around the `logLevelValid` flag and you could print the error as you encounter it.

It would be good to get this code back into a initConfig function.

File: config.js [hunk 8]
```diff
@@ -90,14 +90,14 @@ module.exports = {
     // projects.
     ignoreContextHeader: false,
 
+    // The contents of a key file. If this field is set, its contents will be
+    // used for authentication instead of your application default credentials.
+    credentials: null,
+
```
Comment (ofrobots) [inline config.js]: Why is this change here? It don't seem to to have anything to do with the rest of the changes in this PR.

Comment (kjin) [inline config.js]: The authentication library used in `cloud-common` checks `credentials` and `keyFilename` in reverse order to ours.

Comment (ofrobots) [inline config.js]: Ah, I missed that part. Thanks for clarifying.

File: index.js (hunk 6 above)
Comment (kjin) [inline index.js]: Done

File: index.js (hunk 5 above)
Comment (kjin) [inline index.js]: Done.

File: index.js (hunk 7 above)
Comment (kjin) [inline index.js]: OK, I've used console.error instead.

**Comment by kjin** [comment]
Appveyor test failures seem to be unrelated to these changes.
